txpool/txpool.go

```diff
@@ -555,38 +567,60 @@ func (t *TxPool) Underpriced(tx *types.Transaction) bool {
 	return underpriced
 }
 
-func (t *TxPool) Discard(slots uint64, force bool) ([]*types.Transaction, bool) {
+func (t *TxPool) Discard(remaining uint64, force bool) ([]*types.Transaction, bool) {
 	dropped := make([]*types.Transaction, 0)
-	for t.remoteTxns.Length() > 0 && slots > 0 {
+	for t.remoteTxns.Length() > 0 && remaining > 0 {
 		tx := t.remoteTxns.Pop()
 		dropped = append(dropped, tx.tx)
 
-		txSlots := numSlots(tx.tx)
-		if slots >= txSlots {
-			slots -= txSlots
+		txSlots := slotsRequired(tx.tx)
+		if remaining >= txSlots {
+			remaining -= txSlots
 		} else {
-			slots = 0
+			remaining = 0
 		}
 	}
 
 	// Put back if couldn't make required space
-	if slots > 0 && !force {
+	if remaining > 0 && !force {
 		for _, tx := range dropped {
 			t.remoteTxns.Push(tx)
 		}
 		return nil, false
 	}
+
 	return dropped, true
 }
 
-// increaseSlots increases number of taken slots
-func (t *TxPool) increaseSlots(slots uint64) {
-	atomic.AddUint64(&t.slots, slots)
-}
+// Checks if the incoming tx would cause an overflow
+// and attempts to allocate space for it
+func (t *TxPool) processSlots(tx *types.Transaction, isLocal bool) error {
+	if t.gauge.height+slotsRequired(tx) <= t.gauge.limit {
+		return nil // no overflow
+	}
+
+	if !isLocal && t.Underpriced(tx) {
```

Comment:
I don't think this check here is related to slots themselves, and as such shouldn't be in the processSlots method.

Reply:
It was discussed in @brkomir 's third conversation (now resolved). 

If you still feel it should be moved elsewhere, I can do so.

Reply:
@zivkovicmilos @dbrajovic I understand that you want to avoid adding extra Lines of Code to the addImpl() itself, but we don't extract code into methods/functions because we want to hide things under the carpet, but because we want to contextually organize in this case.

My opinion is that in the long run, this will make it more easier to refactor since we will be aware of this piece of logic, and not put it away in the processSlots.

Reply:
We have discussed this on a huddle, this check for Underpriced needs to occur only if we see there's not enough space in the current slots, that's why it should be in the processSlots. Thank you @dbrajovic for explaining.

@dbrajovic You can perhaps add a comment of similar contents: `If we need to find space for this tx, first let's check if its' gasPrice is less than the tx with the smallest gasPrice in the current txPool`

Reply:
[Added comments](https://github.com/0xPolygon/polygon-sdk/pull/213/commits/6a55bb60673518b5e8ff4cbb4a1c7d768e451999)